components/director/internal/oathkeeper/validation_hydrator_test.go

```diff
@@ -0,0 +1,321 @@
+package oathkeeper
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/persistence/txtest"
+
+	connector "github.com/kyma-incubator/compass/components/connector/pkg/oathkeeper"
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	mocks "github.com/kyma-incubator/compass/components/director/internal/oathkeeper/automock"
+	"github.com/kyma-incubator/compass/components/director/internal/tokens"
+	timeMock "github.com/kyma-incubator/compass/components/director/pkg/time/automock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	token                  = "tokenValue"
+	csrTokenExpiration     = time.Duration(100)
+	appTokenExpiration     = time.Duration(100)
+	runtimeTokenExpiration = time.Duration(100)
+)
+
+func TestValidationHydrator_ResolveConnectorTokenHeader(t *testing.T) {
```

Comment:
instead of asserting just the response code we should also validate the returned auth session (see the previous tests in the controller) and header:
```
var authSession AuthenticationSession
err = json.NewDecoder(rr.Body).Decode(&authSession)
require.NoError(t, err)

// assert.Equal(t, []string{clientId}, authSession.Header[ClientIdFromTokenHeader])
// OR
// assert.Equal(t, emptyAuthSession(), authSession)
```